tests/multiagent/a2a/test_server.py

```diff
@@ -0,0 +1,431 @@
+    """Test that agent_skills handles tools with missing description gracefully."""
+    # Mock tool registry with tool missing description
+    mock_tool_config = {
+        "incomplete_tool": {
+            "name": "incomplete_tool"
+            # Missing description
+        }
+    }
+    mock_strands_agent.tool_registry.get_all_tools_config.return_value = mock_tool_config
+
+    # This should raise a KeyError during initialization when trying to access missing description
+    with pytest.raises(KeyError):
+        A2AServer(mock_strands_agent, skills=None)
+
+
+def test_agent_skills_handles_missing_name(mock_strands_agent):
+    """Test that agent_skills handles tools with missing name gracefully."""
+    # Mock tool registry with tool missing name
+    mock_tool_config = {
+        "incomplete_tool": {
+            "description": "A tool without a name"
+            # Missing name
+        }
+    }
+    mock_strands_agent.tool_registry.get_all_tools_config.return_value = mock_tool_config
+
+    # This should raise a KeyError during initialization when trying to access missing name
+    with pytest.raises(KeyError):
+        A2AServer(mock_strands_agent, skills=None)
+
+
+def test_agent_skills_setter(mock_strands_agent):
+    """Test that agent_skills setter works correctly."""
+    from a2a.types import AgentSkill
+
+    # Mock tool registry for initial setup
+    mock_tool_config = {"test_tool": {"name": "test_tool", "description": "A test tool"}}
+    mock_strands_agent.tool_registry.get_all_tools_config.return_value = mock_tool_config
+
+    a2a_agent = A2AServer(mock_strands_agent, skills=None)
+
+    # Initially should get skills from tools
+    initial_skills = a2a_agent.agent_skills
+    assert len(initial_skills) == 1
+    assert initial_skills[0].name == "test_tool"
+
+    # Set new skills using setter
+    new_skills = [
+        AgentSkill(name="new_skill", id="new_skill", description="A new skill", tags=["new"]),
+        AgentSkill(name="another_new_skill", id="another_new_skill", description="Another new skill", tags=[]),
+    ]
+
+    a2a_agent.agent_skills = new_skills
+
+    # Verify skills were updated
+    assert a2a_agent.agent_skills == new_skills
+    assert len(a2a_agent.agent_skills) == 2
+    assert a2a_agent.agent_skills[0].name == "new_skill"
+    assert a2a_agent.agent_skills[1].name == "another_new_skill"
+
+
+def test_get_skills_from_tools_method(mock_strands_agent):
+    """Test the _get_skills_from_tools method directly."""
+    # Mock tool registry with multiple tools
+    mock_tool_config = {
+        "calculator": {"name": "calculator", "description": "Performs basic mathematical calculations"},
+        "weather": {"name": "weather", "description": "Gets current weather information"},
+    }
+    mock_strands_agent.tool_registry.get_all_tools_config.return_value = mock_tool_config
+
+    a2a_agent = A2AServer(mock_strands_agent, skills=None)
+    skills = a2a_agent._get_skills_from_tools()
```

Comment:
Not for this PR, but possibly a follow-up.  Should tools be able to specify metadata, one of which is tags and/or skills.

We've had other requests for something similar (specifically being able to denote a tool as ReadOnly or Read/Write) and I'm wondering if this jives with that idea